cli/src/pcluster/resources/custom_resources/OpenSearch_lambda/Lambda_pushing_events_to_OpenSearch.py

```diff
@@ -59,6 +77,61 @@ def transform(payload):
     return bulk_request_body
 
 
+def get_node_list(node_names):
+    """
+    Convert node_names to a list of nodes.
+
+    Example input node_names: "queue1-st-c5xlarge-[1,3,4-5],queue1-st-c5large-20"
+    Example output [queue1-st-c5xlarge-1, queue1-st-c5xlarge-3, queue1-st-c5xlarge-4, queue1-st-c5xlarge-5,
+    queue1-st-c5large-20]
+    """
+    matches = []
+    if type(node_names) is str:
+        matches = re.findall(r"((([a-z0-9\-]+)-(st|dy)-([a-z0-9\-]+)-)(\[[\d+,-]+\]|\d+))(,|$)", node_names)
+        # [('queue1-st-c5xlarge-[1,3,4-5]', 'queue1-st-c5xlarge-', 'queue1', 'st', 'c5xlarge', '[1,3,4-5]'),
+        # ('queue1-st-c5large-20', 'queue1-st-c5large-', 'queue1', 'st', 'c5large', '20')]
+    node_list = []
+    if not matches:
+        print("Invalid Node Name Error")
+    for match in matches:
+        node_name, prefix, _, _, _, nodes, _ = match
+        if "[" not in nodes:
+            # Single node name
+            node_list.append(node_name)
+        else:
+            # Multiple node names
+            try:
+                node_range = convert_range_to_list(nodes.strip("[]"))
+            except ValueError:
+                print("Invalid Node Name Error")
+            node_list += [prefix + str(n) for n in node_range]
```

Comment:
This works, but would it not be simpler to iterate through the array of string in the node property rather than try to parse the node_list string?

Reply:
The node property doesn't work that way